packages/@ama-sdk/client-fetch/src/fetch-plugin.ts

```diff
@@ -0,0 +1,46 @@
+import type { ApiClient ,Plugin, PluginAsyncRunner, PluginContext, RequestOptions } from '@ama-sdk/core';
+
+/** Fetch Call Response type */
+export type FetchCall = Promise<Response>;
+
+/**
+ * Interface of an SDK reply plugin.
+ * The plugin will be run on the reply of a call
+ */
+export interface FetchPluginContext extends PluginContext {
+  /** URL targeted */
+  url: string;
+
+  /** Fetch call options */
+  options: RequestInit | RequestOptions;
+
+  /** List of loaded plugins apply to the fetch call */
+  fetchPlugins: PluginAsyncRunner<Response, FetchCall>[];
+
+  /** Api Client processing the call the the API */
+  apiClient: ApiClient;
+
+  // TODO Now supported for all the modern browsers - should become mandatory in @ama-sdk/core@11.0
+  /** Abort controller to abort fetch call */
+  controller?: AbortController;
+}
+
+/**
+ * Interface of an async plugin starter
+ */
+export interface PluginAsyncStarter {
+  /** Determine if the action can start */
+  canStart?(): boolean | Promise<boolean>;
```

Comment:
Question for my personal knowledge
is there a difference with 
```suggestion
  canStart?: () => boolean | Promise<boolean>;
```

Reply:
I copy pasted it from the core actually so I did not check the code.
I wonder if the diff is that this notation enforce the usage of class property.